commit 21593938086fa2886fe07d06aaeb674254a02e3f
Author: Luca Ceresoli <luca.ceresoli@bootlin.com>
Date:   Wed Jan 29 16:19:26 2020 +0100

    docs: i2c: sort index logically
    
    The index page currently lists sections in alphabetical file order without
    caring about their content. Sort sections based on their content logically,
    according to the following structure:
    
     * Intro to I2C/SMBus and their usage in Linux: summary, i2c-protocol,
       smbus-protocol, instantiating-devices, busses/index, i2c-topology,
       muxes/i2c-mux-gpio
     * Implementing drivers: writing-clients, dev-interface,
       dma-considerations, fault-codes, functionality
     * Debugging: gpio-fault-injection, i2c-stub
     * Slave I2C: slave-interface, slave-eeprom-backend
     * Advanced: ten-bit-addresses
     * Obsolete info: upgrading-clients, old-module-parameters
    
    Signed-off-by: Luca Ceresoli <luca@lucaceresoli.net>
    Reviewed-by: Jean Delvare <jdelvare@suse.de>
    Signed-off-by: Wolfram Sang <wsa@the-dreams.de>

diff --git a/Documentation/i2c/index.rst b/Documentation/i2c/index.rst
index a0fbaf6d0675..fee4744475df 100644
--- a/Documentation/i2c/index.rst
+++ b/Documentation/i2c/index.rst
@@ -4,30 +4,66 @@
 I2C/SMBus Subsystem
 ===================
 
+Introduction
+============
+
 .. toctree::
    :maxdepth: 1
 
+   summary
+   i2c-protocol
+   smbus-protocol
+   instantiating-devices
+   busses/index
+   i2c-topology
+   muxes/i2c-mux-gpio
+
+Writing device drivers
+======================
+
+.. toctree::
+   :maxdepth: 1
+
+   writing-clients
    dev-interface
    dma-considerations
    fault-codes
    functionality
+
+Debugging
+=========
+
+.. toctree::
+   :maxdepth: 1
+
    gpio-fault-injection
-   i2c-protocol
    i2c-stub
-   i2c-topology
-   instantiating-devices
-   old-module-parameters
-   slave-eeprom-backend
+
+Slave I2C
+=========
+
+.. toctree::
+   :maxdepth: 1
+
    slave-interface
-   smbus-protocol
-   summary
+   slave-eeprom-backend
+
+Advanced topics
+===============
+
+.. toctree::
+   :maxdepth: 1
+
    ten-bit-addresses
-   upgrading-clients
-   writing-clients
 
-   muxes/i2c-mux-gpio
+Legacy documentation
+====================
 
-   busses/index
+.. toctree::
+   :maxdepth: 1
+
+   upgrading-clients
+   old-module-parameters
 
 .. only::  subproject and html